api/views/diagnostic.py

```diff
@@ -106,9 +109,18 @@ def _treat_csv_file(self, file):
                 canteen = self._create_canteen_with_diagnostic(row, siret)
                 diagnostics_created += 1
                 canteens[canteen.siret] = canteen
+                if not canteen.city and (canteen.city_insee_code or canteen.postal_code):
+                    # if both city code and postcode are given, use city code and avoid having no location due to mismatch
+                    if canteen.city_insee_code:
+                        locations_csv_str += f"{canteen.siret},{canteen.city_insee_code},\n"
+                    else:
+                        locations_csv_str += f"{canteen.siret},,{canteen.postal_code}\n"
+                    hasLocationsToFind = True
             except Exception as e:
                 for error in self._parse_errors(e, row):
                     errors.append(ImportDiagnosticsView._get_error(e, error["message"], error["code"], row_number))
+        if hasLocationsToFind:
```

Comment:
Pourquoi pas appeler _update_location_data directement dans la ligne 118 (et donc sans avoir besoin d'utiliser `hasLocationsToFind` ?

Reply:
Pour qu'on appel l'API une fois, et pas pour chaque ligne (118 est dans le for loop et 122 après)